contracts/marketplace/direct-listings/DirectListingsLogic.sol

```diff
@@ -511,46 +512,69 @@ contract DirectListingsLogic is IDirectListings, ReentrancyGuardLogic, ERC2771Co
         uint256 _totalPayoutAmount,
         Listing memory _listing
     ) internal {
-        (address platformFeeRecipient, uint16 platformFeeBps) = PlatformFeeLogic(address(this)).getPlatformFeeInfo();
-        uint256 platformFeeCut = (_totalPayoutAmount * platformFeeBps) / MAX_BPS;
-
-        uint256 royaltyCut;
-        address royaltyRecipient;
-
-        // Distribute royalties. See Sushiswap's https://github.com/sushiswap/shoyu/blob/master/contracts/base/BaseExchange.sol#L296
-        try IERC2981(_listing.assetContract).royaltyInfo(_listing.tokenId, _totalPayoutAmount) returns (
-            address royaltyFeeRecipient,
-            uint256 royaltyFeeAmount
-        ) {
-            if (royaltyFeeRecipient != address(0) && royaltyFeeAmount > 0) {
-                require(royaltyFeeAmount + platformFeeCut <= _totalPayoutAmount, "fees exceed the price");
-                royaltyRecipient = royaltyFeeRecipient;
-                royaltyCut = royaltyFeeAmount;
+        address _nativeTokenWrapper = nativeTokenWrapper;
+        uint256 amountRemaining;
+
+        // Payout platform fee
+        {
+            (address platformFeeRecipient, uint16 platformFeeBps) = PlatformFeeLogic(address(this)).getPlatformFeeInfo();
+            uint256 platformFeeCut = (_totalPayoutAmount * platformFeeBps) / MAX_BPS;
+
+            // Transfer platform fee
+            CurrencyTransferLib.transferCurrencyWithWrapper(
+                _currencyToUse,
+                _payer,
+                platformFeeRecipient,
+                platformFeeCut,
+                _nativeTokenWrapper
+            );
+
+            amountRemaining = _totalPayoutAmount - platformFeeCut;
+        }
+
+
+        // Payout royalties
+        {
+            // Get royalty recipients and amounts
+            (address payable[] memory recipients, uint256[] memory amounts) = RoyaltyPaymentsLogic(address(this))
```

Comment:
Let's use the general/widespread recognizable `IRoyaltyEngineV1` interface to make the call, instead of our own custom `RoyaltyPaymentsLogic`

Reply:
The code in `getRoyalty` of RoyaltyPaymentsLogic handles the case when royalty-engine is not present on some network (needed for any evm). 
Hence, I guess using `IRoyaltyEngine` interface to make the call will be confusing to read, since we're not directly calling the engine, and marketplace-contract is not an engine either.